### DISCUSSION TITLE
`fi.floris.turbine_map` equivalent in v3

### DISCUSSION THREAD

**Discussion by vijay092** [NatLabRockies/floris#320] [Q&A] [answered]
Looks like `fi.floris.turbine_map` has been removed from FLORIS 3.0. Is there an equivalent in the new version?
I would like to perform the following operation:
```

for coord, turbine in fi.floris.farm.turbine_map.items:
        ct.append(turbine.Ct)
        ai.append(turbine.aI)
        ave_vel.append(turbine.average_velocity)

```

**Comment by rafmudaf** [comment, marked as answer]
Hi @vijay092 

In v3, we've moved from having a instance of the `Turbine` class for each turbine in the wind plant to having a turbine definition and using that along with the conditions at the turbine to calculate the performance quantities. You can use the functions in [src/floris/simulation/turbine.py](https://github.com/NREL/floris/blob/v3/src/floris/simulation/turbine.py) for [power](https://github.com/NREL/floris/blob/v3/src/floris/simulation/turbine.py#L80), [Ct](https://github.com/NREL/floris/blob/v3/src/floris/simulation/turbine.py#L137), [axial induction](https://github.com/NREL/floris/blob/v3/src/floris/simulation/turbine.py#L174), and [average velocity](https://github.com/NREL/floris/blob/v3/src/floris/simulation/turbine.py#L210). Moving these out of the turbine class allows for a much more efficient computation.

We don't have a formal example for this yet, but a pattern is included in the regression tests:
https://github.com/NREL/floris/blob/v3/tests/reg_tests/jensen_jimenez_regression_test.py#L111

Does this answer your question?

**Reply by vijay092** [reply]
Great, thanks so much for the clarification!

**Reply by rafmudaf** [reply]
No problem, thank you for using v3! Please keep in touch if you have other questions or feedback.

**Comment by vijay092** [comment]
In the current implementation, is it possible for every turbine in the farm to have different parameters?

**Reply by bayc** [reply]
I am currently implementing that and will have it merged soon! By different parameters, you mean hub height, rotor diameter, Cp/Ct table, those kind of parameters, right?

**Reply by vijay092** [reply]
Hey @bayc! Yes, that is correct. Looking forward to it :)
